### ISSUE TITLE
ROS-M: Use std::shared_ptr

### ISSUE THREAD

**Issue by davetcoleman** [moveit/srdfdom#33]
Currently this repo has:
```
typedef boost::shared_ptr<Model> ModelSharedPtr;
typedef boost::shared_ptr<const Model> ModelConstSharedPtr;
```
Unlike the rest of moveit

**Comment by v4hn** [comment]
could we still change that in -L already?

**Comment by davetcoleman** [comment]
I don't think we can change this in Lunar until we decide we want to branch all of MoveIt! to Lunar, which I'm ok with doing now that we've discontinued Jade. Let's discuss at a maintainer meeting.

**Comment by v4hn** [comment]
Hm, did we forgot to talk about this in the meeting?

**Comment by davetcoleman** [comment]
It was in my notes for the meeting and I overlooked it, yea. I don't think this nit-picky cleanup is motivation to branch for L yet, but I certainly hope to see this in M.

**Comment by rhaschke** [comment]
Resolved in #36.
